### ISSUE TITLE
Icinga2 - Can't login with AD-Account that has access to IcingaWeb2

### ISSUE THREAD

**Issue by RuvenG** [HenriWahl/Nagstamon#278]
Good morning ladies and gentlemen,

I just tried to use the Icinga2 Icingaweb2-GUI with the new Nagstamon beta-Client (newest Version) and encountered a litle problem. I might not even be a problem, maybe I just did something wrong.

So here's what has happened:

I installed the Nagstamon 64bit Client on a Windows 8.1-System and filled in the server credentials.

As for "Monitor URL" I used the IcingaWeb2 Main page: "http://icingaserver/icingaweb2"

And as username I chose to pick an AD-User that has admin access to Icinga2/ Icingaweb2.

The problem however is that when I want to confirm my input, Nagstamon just closes down completely and nothing happens. When I now try to reopen Nagstamon it will say that it is missing a configuration.

Funny thing is when I intentionally type in a wrong password/user Nagstamon will start working properly but will eventually show "authentication failed", which is pretty much the expected output.

Now I am not quite sure if it's a bug or if I'm actually screwing something up here but I thought I should let you know.

Thanks.

Greetings

**Comment by HenriWahl** [comment]
Which version exactly?

**Comment by RuvenG** [comment]
Nagstamon-2.0-beta-20160729-win64_setup.exe

I downloaded this version.

**Comment by HenriWahl** [comment]
Can you enable debugging output in settings and post it? Maybe there will be a hint.

**Comment by RuvenG** [comment]
I enabled debugging but nagstamon doesn't write anything to the log when problem appears.

I just testet a stable version (1.0.1) with our older CheckMK system and the same error occured. After entering the password and trying to connect to the server, the whole process disappeared.

I did also test this with an older Version (0.9.7.1) and it worked perfectly fine.

**Comment by HenriWahl** [comment]
IcingaWeb2  can't work with Nagstamon 0.9.7.1... what OS do you run it on?

**Comment by RuvenG** [comment]
We are running Icinga2 on Ubuntu 14.4.

And like I said, I tested CheckMK with with either 1.0.1 and 0.9.7.1 because I was curious if it worked in general or if there is a totally different problem. I then noticed that I have pretty much the same problem on 1.0.1 with CheckMK as I do with Icinga2 on 2.0. CheckMK works fine with 0.9.7.1 however.

**Comment by HenriWahl** [comment]
And Nagstamon? Which OS do you use?

**Comment by RuvenG** [comment]
I mentioned it in the original post;

Windows 8.1 64Bit

**Comment by HenriWahl** [comment]
I saw right now... sorry :-)

**Comment by HenriWahl** [comment]
Look somehow like authorization trouble... do you use the system keyring for storing the passwords in the Nagstamon settings?

**Comment by RuvenG** [comment]
Ok, little update.

I just installed Version 2.0 ( from 0729) again to check if the CheckMK-Monitoring system works.

And yes, the CheckMK System in combination with Nagstamon 2.0 works.

Icinga2 however still crashes when I enter the correct username and password. Again when i enter a wrong passwd/user-combo it won't crash but tell me authentication failure.

According to your question, what exactly do you mean by "system keyring", probably the "system keychain2? Under Windows 8 its called Vault, but no, Im not using any of those mechanics.

When I try to log on the Icinga-Interface the Nagstamon-Process just disappears.

**Comment by HenriWahl** [comment]
Yes, this thingy called vault. Came in as option some versions ago - see on first tab of settings dialog,

**Comment by RuvenG** [comment]
Ok, didn't even see it's in there now.

It was off originally but i tried with the feature turned on and off, same result.

**Comment by HenriWahl** [comment]
Can you run it from source in a console to get some crash message?

**Comment by RuvenG** [comment]
sorry for not responding yesterday, had some troubled server that needed some work.

Since I am using Windows I can't really run it from source like I would under Linux.

According to the Windows_logs there is no crash whatsoever, there isn't even a warning or something similar. 

I actually don't think that it it worth to investigate this any further at this point, I am going to do some further tests and then I come back to you with more details.


Thanks for your help.

**Comment by RuvenG** [comment]
Ok,

i just got it to work.

I can't actually tell why I wasn't able to sort this out yesterday since I already tried what I just did.

Pretty much all I did was to use FQDN for the Icinga2-Server and to tell the Nagstamon Client NOT to use cookie based authentication and it suddenly worked flawlessly... seems like i screwed something up while trying this yesterday. Anyway Im happy now!

Thank you for your help, very much appreciated!

Have a good day, bye.
